### PR TITLE
feat(apple): Add tracePropagationTarget to automatic instrumentation

### PR DESCRIPTION
**Pull request by brustolin**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Added an explanation on how to filter out requests that will contain the trace id by using 'tracePropagationTarget' option.

In response to #7110

### CODE REVIEW

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 1]
```diff
@@ -161,6 +161,34 @@ SentrySDK.start { options in
 [NSURLSession]: https://developer.apple.com/documentation/foundation/nsurlsession
 [NSURLConnection]: https://developer.apple.com/documentation/foundation/nsurlconnection
 
+### Trace Propagation Targets
+
+Sentry addes an extra header with the trace id in the outgoing HTTP requests in order to continue the transaction in the backend.
```
Comment (shanamatthews) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:166]:
```suggestion
Sentry adds an extra header with the trace id in the outgoing HTTP requests to continue the transaction in the backend.
```

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 2]
```diff
@@ -161,6 +161,34 @@ SentrySDK.start { options in
...
+
+It's possible to filter out which requests Sentry should add the extra header by setting the `tracePropagationTarget` option, this way, only the app backend will receive the trace id.
```
Comment (shanamatthews) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:168]:
```suggestion
You can set the `tracePropagationTarget` option to filter which requests Sentry adds the extra header to.
For example, to ensure that only the app backend will receive the trace id.
```

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 3]
```diff
@@ -161,6 +161,34 @@ SentrySDK.start { options in
...
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    options.tracePropagationTargets = @[@"MyAppDomain.com"]
+}];
+```
+
+The option may contain a list of NSString or NSRegularExpression against which the URLs of outgoing requests are matched. If one of the entries in the list matches the URL of an outgoing request, trace data will be attached to that request. String entries do not have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.
+
+If tracePropagationTargets is not provided, trace data is attached to every outgoing request from the instrumented client.
```
Comment (shanamatthews) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:190]:
```suggestion
If `tracePropagationTargets` is not provided, trace data is attached to every outgoing request from the instrumented client.
```

**Review by shanamatthews — APPROVED** [review]

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 4]
```diff
@@ -161,6 +161,35 @@ SentrySDK.start { options in
 [NSURLSession]: https://developer.apple.com/documentation/foundation/nsurlsession
 [NSURLConnection]: https://developer.apple.com/documentation/foundation/nsurlconnection
 
+### Trace Propagation Targets
+
+Sentry adds an extra header with the trace id in the outgoing HTTP requests to continue the transaction in the backend.
+
+You can set the `tracePropagationTarget` option to filter which requests Sentry adds the extra header to.
+For example, to ensure that only the app backend will receive the trace id.
```
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:169]: `l`:
```suggestion
For example, to ensure that only your app backend will receive the trace id.
```

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 5]
```diff
@@ -161,6 +161,35 @@ SentrySDK.start { options in
...
+```objc {tabTitle:Objective-C}
+@import Sentry;
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    options.tracePropagationTargets = @[@"MyAppDomain.com"]
+}];
+```
+
+The option may contain a list of NSString or NSRegularExpression against which the URLs of outgoing requests are matched. If one of the entries in the list matches the URL of an outgoing request, trace data will be attached to that request. String entries do not have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.
```
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:189]: `l`:
```suggestion
The option may contain a list of `NSString` or `NSRegularExpression` against which the URLs of outgoing requests are matched. If one of the entries in the list matches the URL of an outgoing request, trace data will be attached to that request. String entries do not have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.
```

**Pull request merged**